### PR TITLE
feat (Minimal) : Add `EmptyResult`

### PR DESCRIPTION
**Pull request by ShreyasJejurkar**
1. Add `EmptyResult`. 
2. Add corresponding tests.

Addresses https://github.com/dotnet/aspnetcore/issues/40168

Opening it early for feedback. 

Waiting for API approval, for changes!

### CODE REVIEW

File: src/Http/Http.Results/src/Results.cs
Comment (Kahbazi) [inline src/Http/Http.Results/src/Results.cs:729]:
```suggestion
    public static IResult Empty() => EmptyResult.Instance;
```

File: src/Http/Http.Results/src/EmptyResult.cs
Comment (Kahbazi) [inline src/Http/Http.Results/src/EmptyResult.cs:13]:
```suggestion

    private EmptyResult()
    {
    }
    
    public static EmptyResult Instance { get; } = new EmptyResult();
    
```

**Review by Kahbazi — COMMENTED** [review]
How about creating only one instance of `EmptyResult` and reuse it?

File: src/Http/Http.Results/src/EmptyResult.cs
Comment (davidfowl) [inline src/Http/Http.Results/src/EmptyResult.cs:18]:
```suggestion
        return Task.CompletedTask;
```

File: src/Http/Http.Results/src/EmptyResult.cs
Comment (BrennanConroy) [inline src/Http/Http.Results/src/EmptyResult.cs:14]: Add a private constructor so you can't new one of these up from outside the class.

File: src/Http/Http.Results/test/EmptyResultTest.cs
Comment (halter73) [inline src/Http/Http.Results/test/EmptyResultTest.cs:15]: Can we assert a 200 status and that the Body MemoryStream hasn't been written to?

**Review by halter73 — APPROVED** [review]
LGTM pending API approval. I would like a few more asserts in the test though.

File: src/Http/Http.Results/test/EmptyResultTest.cs
Comment (campersau) [inline src/Http/Http.Results/test/EmptyResultTest.cs:10]: Does this need a `[Fact]` to declare it as an actual test?

```suggestion
    [Fact]
    public async Task EmptyResult_DoesNothing()
```

Comment (halter73) [reply src/Http/Http.Results/test/EmptyResultTest.cs:10]: Good eye!

Comment (ShreyasJejurkar) [reply src/Http/Http.Results/test/EmptyResultTest.cs:11]: Added in the latest commit. Thanks for the catch! But still, the test did execute without this attribute! I wonder!

File: src/Http/Http.Results/test/EmptyResultTest.cs
Comment (ShreyasJejurkar) [reply src/Http/Http.Results/test/EmptyResultTest.cs]: Compared status code. Not sure if I compare streams correctly!?

File: src/Http/Http.Results/test/EmptyResultTest.cs
Comment (halter73) [inline src/Http/Http.Results/test/EmptyResultTest.cs:23]:
```suggestion
        var memoryStream = (MemoryStream)httpContext.Response.Body;
        await emptyResult.ExecuteAsync(httpContext);

        // Assert
        Assert.Equal(StatusCodes.Status200OK, httpContext.Response.StatusCode);
        Assert.Equal(0, memoryStream.Length);
```

It might be slightly better to just make the `Response.Body` stream a parameter to `GetHttpContext(Stream)`. That way you could avoid casting back to `MemoryStream`, but it's not a big deal either way.

File: src/Http/Http.Results/test/EmptyResultTest.cs
Comment (halter73) [reply src/Http/Http.Results/test/EmptyResultTest.cs]: Looks good.

**Comment by BrennanConroy** [comment]
@ShreyasJejurkar The API has been approved per comment https://github.com/dotnet/aspnetcore/issues/40168#issuecomment-1054550507, please update as appropriate 😃

**Comment by ShreyasJejurkar** [comment]
> @ShreyasJejurkar The API has been approved per comment [#40168 (comment)](https://github.com/dotnet/aspnetcore/issues/40168#issuecomment-1054550507), please update as appropriate 😃

Yup, updated with the latest commit. I wasn't subscribed to the issue, so was not aware! 

BTW do we need to change it for ControllerBase as well? MVC as well? That wasn't proposed!

**Review by davidfowl — APPROVED** [review]
LGTM

File: src/Http/Http.Results/src/Results.cs
Comment (Kahbazi) [inline src/Http/Http.Results/src/Results.cs:727]:
```suggestion
    /// Produces an empty result response, that when executed will do nothing.
```

Comment (Kahbazi) [inline src/Http/Http.Results/src/Results.cs:727]:
```suggestion
    /// Produces an empty result response, that when executed will do nothing.
```

**Comment by ShreyasJejurkar** [comment]
> How about creating only one instance of `EmptyResult` and reuse it?

Yes, we are doing exactly that in here! :)

**Comment by davidfowl** [comment]
Thanks @ShreyasJejurkar !

**Comment by ShreyasJejurkar** [comment]
> Thanks @ShreyasJejurkar !

I am glad I got to contribute and work with you all people. Thanks! :)

**Pull request merged**
